### ISSUE TITLE
Improve mission feedback

### ISSUE THREAD

**Issue by AndreasAntener** [mavlink/qgroundcontrol#2715]
Mission states are reported via various messages. One problem on the PX4 side is that the message severity sometimes make no sense. The other issue is that in QGC you need to check the messages.

A few problematic cases should be handled better, e.g. mission not accepted, takeoff too far away, etc. I'm not actually sure what is done how at the moment in terms of notifications.

@sanderux

**Comment by sanderux** [comment]
Thnx, i'll see if i can improve this.

**Comment by AndreasAntener** [comment]
@DonLakeFlyer @LorenzMeier do the message severity level currently play a role in communication with QGC?

**Comment by LorenzMeier** [comment]
Yes, the severity controls what is spoken and what not. Anything at critical or higher will be spoken.

**Comment by AndreasAntener** [comment]
Oh, wasn't there that thing with the # (hash) in front of the message for that? For the FW stuff we used `mavlink_log_info(_mavlink_fd, "#Takeoff on runway");` and I think QGC speaks that as well

**Comment by LorenzMeier** [comment]
That's deprecated.

**Comment by DonLakeFlyer** [comment]
The message severity determine whether the toolbar icon turns to a yield sign. Also messages above a certain severity cause the message to show in a popup at the bottom of the screen. Unless the severity trigger threshold is incorrect in QGC (or you think the toolbar/popup mechanism is insufficient) the changes needed to make this better will come from the firmware side, not the QGC side.

**Comment by sanderux** [comment]
My proposal for mission feedback and analysis would be the following;

1. Revisit the mavlink log feedback. Currently a mission always raises a critical alert that states the takeoff altitude. This results in less attention to true critical alerts that could contain safety or feasibility warnings.
2. There is currently no GCS feedback on the result of uploading a mission to the FC. I have recently experienced confusion that resulted from an apparent failure to send the mission to the FC. I would propose that QGC indicate both success and failure of sending the mission. The latter i think should be a very clear warning.
3. While tracking a mission through radio telemetry it would provide a lot of clarity to see which mission item is currently being executed. This could simply be provided as mavlink log messages (info), but ideally it would be displayed in the "FLY" window of QGC. I think it now changes the color on the current mission item but in the field this is very hard to see. If the mission list would be displayed as it is in the "PLAN" window, and it would follow the currently executed item, this would provide more clarity. Not only on the sequence number, but also what that command actually is.
4. Post flight analysis of a mission can now only be performed if telemetry was connected and saved from QGC. From the onboard logs there is no way to tell (AFAIK) what mission item was being executed, or even what the mission was. I would propose that the mission is added to the preflight log, and the current mission item index added to the flight log. If mid-flight mission changes should also be considered perhaps a separate mission log is an option. 

I realize points 1 and 4 are on the PX4 side, and 2/3 on QGC side but this would be my take on the user perspective of mission feedback and analysis.

**Comment by DonLakeFlyer** [comment]
#2 QGC will display an error if send to vehicle flails, with details of what went wrong. Nothing currently on a successful send.

**Comment by LorenzMeier** [comment]
There should be no notification on successful sends. That's meaningless and does not represent modern user interface design. If something goes wrong it will tell you.

**Comment by AndreasAntener** [comment]
I generally agree Lorenz, but in this case it should at least show the upload progress. Ok to be fair, I hear the buzzer when it's accepted and the check passed. But if I upload a big mission and don't have the buzzer (or the plane is already in the air) I have no notion of what's going on.

**Comment by DonLakeFlyer** [comment]
The mission controller has an in progress signal internally. My original plan was to animate/rotate the sync icon in the button when send/recevie was in progress. I can change that when I get back.

**Comment by sanderux** [comment]
@DonLakeFlyer 
From the dev call we also got the request to support user configurable severity for spoken messages.

**Comment by DonLakeFlyer** [comment]
> request to support user configurable severity for spoken messages.

Seems interesting, but worried that this really is being requested because the firmware isn't marking severity appropiately. Hence users want to tone town all the talking the firware is causing. Do you have a use case where changing the setting for this is a useful feature? Some example message traffic to show how changing back/forth is helpful?

From my standpoint, adding an option is always a last resort instead a correct default.

**Comment by sanderux** [comment]
The request was made in light of people wanting spoken info messages when flying FPV.

**Comment by DonLakeFlyer** [comment]
Ah, makes perfect sense, thanks.
